### ISSUE TITLE
Failure to launch on iOS 12.5.7

### ISSUE THREAD

**Issue by brian-farrell** [omz/Pythonista-Issues#693]
The information in the App Store says Pythonista 3 requires iOS 12.0 or later, but the app fails to launch on iOS 12.5.7.

**Comment by cclauss** [comment]
Is there any error message or does it just start and stop right away?

Please try opening your browser and going to the URL `pythonista3://`

**Comment by omz** [comment]
Unfortunately, it seems that 3.4 doesn't work on iOS 12, due to one included framework accidentally being compiled with an iOS 14 deployment target.

I'm looking into this, but it's unfortunately become pretty difficult to test on iOS 12, as I currently don't have a device that runs it, and the iOS 12 simulator doesn't work on Ventura. But I think I'll figure it out somehow, sorry about this!

**Comment by brian-farrell** [comment]
It sounds like you have a handle on the problem, but just to follow-up with @cclauss :

1. There is no error message displayed.  The app launches very briefly, showing the initial splash screen (I don't seem to be able to get a screenshot of this).  The splash screen  disappears and then I'm returned to the phone's Home screen.  Interestingly, it appears to be running in the background, as it shows-up when I double-click the Home button (I have attached a screenshot of this), but when I select Pythonista from that view, I am immediately returned to the Home screen, just like I am when I initially tried to launch the app.
2. The `pythonista3://` URL behaves differently, depending on which browser I use.  When I enter it into Firefox (my main browser) the URL is not recognized and I am shown Google search results instead.  When I enter the URL into Safari, I am asked if I want to open the page in Pythonista.  When I select `<Open>` from the Safari dialog, I see the same behavior as I do when clicking on the app from the Home screen.

I understand your difficulty with testing on iOS 12.  I'm happy to help if I can.  I've experimented with programming in Swift for iOS in the past, so I have *some* familiarity with the process.  Please let me know if there is anything that I can do to assist from a testing standpoint.  I live just outside of Washington, DC in America, so there's a good bit of a time difference, but I'm also willing to accommodate that as best I can.

I love this app - I wish I could afford a newer iPhone, but I simply can't at the moment.  In any case, I wish you all the best!

![IMG_0150](https://user-images.githubusercontent.com/2137966/235377870-c8ef1faa-ffdd-4a95-8b58-4907f8194149.jpeg)
![IMG_0155](https://user-images.githubusercontent.com/2137966/235377871-0b1d190c-c4f7-4e3e-9d57-13c63e830179.jpeg)
![IMG_0156](https://user-images.githubusercontent.com/2137966/235377872-6e7b4c1c-8c42-4544-be7d-d9274fc9d4a0.jpeg)

**Comment by brian-farrell** [comment]
FWIW - I also have an older Mac, running Big Sur 11.7.6 with Xcode 12.4 and plenty of older simulators...

![Xcode_12-4](https://user-images.githubusercontent.com/2137966/235379133-ece14cb2-f01a-4c56-aaa0-dfbf58046bdc.png)

**Comment by wgaylord** [comment]
Any progress on this. It appeara to be the freetype library according the my crashlog. Mostly use Pythonista on my old Ipad and Old Iphone 7. So sort of hope this gets fixed so I can use it on a big screen again.

**Comment by cclauss** [comment]
@wgaylord Did you try opening `pythonista3://` in your browser?

**Comment by wgaylord** [comment]
Yes Crashes with the same library not loaded error in the crash log.
https://drive.google.com/file/d/1nDxlsvSa4GPBCsEKXPiiMntXsDxZDuSu/view?usp=drivesdk
is my crash log

**Comment by leovanderveen** [comment]
I just purchased Pythonista on my older iPad mini 2, iOS 12.5.7, wish I've read this post earlier since it still crashes upon launching the app. Any news on a solution?

**Comment by wgaylord** [comment]
Any updates?

**Comment by wgaylord** [comment]
Crashlog when using the latest 3.5 beta https://drive.google.com/file/d/1tqcltNi93-JR0aIRj8uJWKnZHgQ9Qi8u/view?usp=drivesdk

App version 350002 on ios 12.5.7

**Comment by omz** [comment]
@wgaylord Is this still crashing on launch? I've tested the beta on an iPhone 5s, running iOS 12.5.7, and it seems to be working fine.

Has anyone else tested the 3.5 beta on iOS 12?

**Comment by brian-farrell** [comment]
Hi - just curious if you have any slots left on TestFlight.  I'd be happy to be a beta tester, if it would help.  I'm still running iOS 12.5.7 on an iPhone 6.

**Comment by yamisama** [comment]
Same issue for me on iPad Air under iOS 12.5.7. There doesn't even seem to be a way to access the script files created with the application.

It's quite a pity since I had hoped that running things with Pythonista on it would still be a good use for the device.

A simple update in the store that returns the app to the last usable state for iOS 12 would be enough.

**Comment by wgaylord** [comment]
Can confirm latest testflight is now working on 12.5.7

**Comment by lunatrop** [comment]
Yea was real excited to see this work on my old IPad 12.5.7 but it only works on my iPhone 8.

Checked the logs and DYLD throws freetype.framework error. I guess that’s why we’re here at issue 693.

Would be lovely to see this up and running soon.

**Comment by yamisama** [comment]
Any information on when that testflight build is estimated to be released through the store? My iPad Air still doesn't show any updates.

**Comment by cclauss** [comment]
iOS 26 release is in a few hours...  https://www.apple.com/os/ios
* https://github.com/omz/Pythonista-Issues/issues/793#issuecomment-3293715357

**Comment by yamisama** [comment]
Sorry for the bump, but are there any news or any rough estimates regarding a release version of the testflight build that has been mentioned?

To be honest I'm a bit concerned that Apple might shut down the App Store for the affected devices before the fix is deployed, permanently locking us out of the app and any scripts developed in it that were not yet backed up before the update that broke it.

**Comment by angrod2eresse** [comment]
I can reproduce this on iPad Air 1 (iPad4,1), iOS 12.5.7.

Pythonista 3.4 from the App Store crashes immediately at launch.

Crash log:
- Version: 3.4
- Build: 340012
- Bundle ID: com.omz-software.Pythonista3
- Hardware Model: iPad4,1
- OS: iPhone OS 12.5.7 (16H81)
- Exception Type: EXC_CRASH (SIGABRT)
- Termination Description:
  DYLD, Library not loaded: @rpath/freetype.framework/freetype
  Referenced from: Pythonista3.app/Frameworks/Py3Kit.framework/Py3Kit
  Reason: freetype.framework/freetype: missing LC_DYLD_INFO load command

Pythonista 3.3 build 330025 works on the same device/OS, so this appears to be a Pythonista 3.4 packaging/framework compatibility issue with iOS 12.5.7, not a device RAM issue.

If Pythonista 3.4 cannot be made compatible with iOS 12, could the App Store minimum OS requirement for 3.4 be raised to iOS 13? That should allow iOS 12 users to download the last compatible version, Pythonista 3.3, from purchase history. This would be much better than leaving iOS 12 users with only a crashing 3.4 build.

[Pythonista3-2026-07-02-113031.txt](https://github.com/user-attachments/files/29581318/Pythonista3-2026-07-02-113031.txt)

**Comment by wgaylord** [comment]
Any progress on this?

**Comment by angrod2eresse** [comment]
I installed AppStore++ and I could downgrade Pyhthonista 3.4 to 3.3.  But in order to install AppStore++, I had to jailbreak my iPad.

**Comment by yamisama** [comment]
@angrod2eresse Would you mind giving the rough steps of how to do this? 

The information I found suggests in order to install AppStore++, I would need to install something called TrollStore, which claims not to require a jailbreak. But it lists iOS 14 Beta 2 or higher as a requirement, whereas my iPad Air is on iOS 12.

**Comment by angrod2eresse** [comment]
Chimera worked for my iPad Air 1/iOS 12.5.7.  I don't know if it works for 12.5.8.
1. Install AltStore, a sideloder on your PC.
2. Install Chimera and perform jailbreak.
3. Install AppStore++ by Sileo.  It works in background and you won't see an icon on your home screen.
4. Open official AppStore and find Pythonista 3.4.  It must be installed.
5. Long tap open and you'll see pull down menu that lets you downgrade.

Remember to disable 'App Updates' in settings, or else you'll see Pythonista updated to 3.4 next morning.

**Comment by yamisama** [comment]
Thank you very much! I will definitely give that a try on my iPad Air if we don't get an official fix soon.
